projects/storefrontlib/shared/components/chat-messaging/avatar/avatar.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ICON_TYPE } from '@spartacus/storefront';
+import { MessageEvent } from '../messaging';
+
+@Component({
+  selector: 'cx-avatar',
+  templateUrl: './avatar.component.html',
+})
+export class AvatarComponent implements OnInit {
+  @Input() event: MessageEvent;
+  iconTypes = ICON_TYPE;
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  getIntitials(name: string) {
+    const names = name.split(' ');
+    return `${names[0]?.split('')[0]}${names[1] ? names[1]?.split('')[0] : ''}`;
```

Comment:
this is pretty hard to read 

what do you think of this apparoach

```
 return name.split(' ') --> to separate any spaced names as that is the norm I believe
     .map(name => name[0]) --> transform the array containing the names to only have the first letter
    .join(''); --> to put the 'array' into a string

```

Reply:
yup, that's better.